SwiftCSV/Parser.swift

```diff
@@ -81,7 +81,11 @@ enum Parser {
 
             guard rowIndex >= offset else { return }
             fields.append(String(field))
-            rowCallback(fields)
+            do {
+                try rowCallback(fields)
+            } catch {
+                throw CSVParseError.generic(message: "Error thrown by block for row \(rowIndex)")
```

Comment:
You're right, this doesn't change usage because the function is throwing anyway.

But could you avoid catching, wrapping, and rethrowing the error? User code throwing a `MyVerySpecificError` on failure in this callback will likely want to work with and maybe even catch `MyVerySpecificError`, not `CSVParseError.generic`, losing the original error's information completely here.

For context: I'm also not a fan of wrapping-and-rethrowing, because that adds another step to the debugging flow when you need to inspect the stack. 

Just let rowCallback's error propagate will do The Right Thing in more cases 👍